File: sort/selectionSort.java
//O(n^2): n + n-1 + n-2 + ..+1
//O(1): for i, j, global
public int[] selectionSort(int[] arr){
	if(arr == null || arr.length <= 1)
		return arr;
	int globalMin;
	for(int i = 0; i < arr.length - 1; i++){
		globalMin = i;
		//find the min element in subarray of (i, array.length-1)
		for(int j = i + 1; j < arr.length; j++){
			if(array[j] < arr[globalMin])
				globalMin = j;
		}
		swap(arr, i, globalMin);
	}
	return arr;
}
public void swap(int[] arr, int left, int right){
	int temp = arr[left];
	arr[left] = arr[right];
	arr[right] = temp;
}

File: sort/selectionSort.py
def selectionSort(arr):
	for i in range(len(A)):  # A=[10,20,100,3, -3]  len = 5, range(5) = 0,1,2,3,4
		min_index = i  # 0 1 2 3 4
		for j in range(i + 1, len(A))
			if A[min_index] > A[j]:
				min_index = j

		# swap(A, i, min_index)
		A[i], A[min_idx] = A[min_idx], A[i]
